File: java/pro.java

import java.util.Scanner;
public class pro{
public static void main(String args[])
{
  int[][] a=new int[3][3];
  int[][] b=new int[3][3];
  int[][] c=new int[3][3];
int i,j,k,sum; 
System.out.println("enter a 3*3 array\n");
Scanner in = new Scanner(System.in);
for(i=0;i<3;i++)
    {
for( j=0;j<3;j++)
{    
a[i][j] = in.nextInt();
}
}
System.out.println("enter a 3*3 array\n");
for(i=0;i<3;i++)
    {
for( j=0;j<3;j++)
{    
b[i][j] = in.nextInt();
}
}
for(i=0;i<3;i++)
  {
   for(j=0;j<3;j++)
   {
        sum = 0;            
    for(k=0;k<=2;k++)
     {
      sum = sum + a[i][k] * b[k][j];
     }
   c[i][j]=sum;
   
   }
  }

  System.out.println("new array after multiplication\n");
 for(i=0;i<3;i++)
  {
   for(j=0;j<3;j++)
   {
    System.out.println("\t"+c[i][j]);
    }
 System.out.println("\n");
}
}
    
}

 

 

File: ruby/pro.rb
def mul(c,d)
m=[3],[3]
for i in 0..3
  for j in 0..3
   sum=0
   for k in 0..3
sum=sum+c[i][k]+d[k][j]
end
m[i][j]=sum
 end
  end
puts"array after multiplication"
for k in 0..3
  for l in 0..3
puts "#{m[k][l]}"
end 
 end
  end

  a=[3],[3]
   b=[3],[3]
 
puts"enter a  array\n"
for row in 0..3
  for col in 0..3
a[row][col]= gets;
end
 end

puts"enter a  array\n"
for row in 0..3
  for col in 0..3
b[row][col] = gets;
end
 end
 mul(a,b)
